packages/protocol-parser/src/decodeValueArgs.ts

```diff
@@ -13,7 +13,7 @@ export function decodeValueArgs<TSchema extends ValueSchema>(
     valueSchema,
     concatHex([
       readHex(staticData, 0, staticDataLength(Object.values(valueSchema).filter(isStaticAbiType))),
-      encodedLengths,
+      padHex(encodedLengths, { size: 32 }),
```

Comment:
do we wanna add a test somewhere that captures the before/after of this change to demonstrate the bug and the fix?